src/context/project-settings.tsx

```diff
@@ -0,0 +1,151 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { FormProps as SpruceFormProps } from "@rjsf/core";
```

Comment:
nit. We can probably reexport this from SpruceForm by default. No preference tho.